app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,9 @@
+class FoodsController < ApplicationController
+
+  def index
+    @facade = FoodDataFacade.new
+    @total_hits = @facade.total_hits(params[:q])
+    @foods = @facade.search_by_ingredient(params[:q]).first(10)
```

Comment:
Ultimately these 2 lines are hitting the same endpoint. You could clean this up by doing something like this in your service:

```ruby
  def search_by_ingredient(query)
    food_data = service.search_by_ingredient(query)
    foods = food_data[:foods].map do |data|
      Food.new(data)
    end
    [foods, food_data[:totalHits]]
  end
  ```
  
  And then in your controller assign like this:
  `@foods, @total_hits = @facade.search_by_ingredient(params[:q]).first(10)`
  
Then you can get rid of the `total_hits` method in the facade altogether as well as make just one call to the API.